charts/datadog/templates/_helpers.tpl

```diff
@@ -655,6 +655,16 @@ Returns env vars correctly quoted and valueFrom respected
 {{- end -}}
 {{- end -}}
 
+{{/*
+Returns env vars correctly quoted defined in a dict
+*/}}
+{{- define "additional-env-dict-entries" -}}
+{{- range $key, $value := . }}
+- name: {{ $key }}
+  value: {{ $value | quote }}
```

Comment:
what about `valueFrom` support? do you think you can also support it?

Reply:
Thanks for the suggestion @clamoriniere! I've added that functionality by using below:

```
{{- if kindIs "map" $value }}
{{ toYaml $value | indent 2 }}
{{- else }}
  value: {{ $value | quote }}
{{- end }}
```

Then you can use either of:

```yaml
envDict:
  MY_STRING_VAR: myValue
  MY_VAL_FROM_VAR:
    valueFrom:
      secretKeyRef:
        key: myKey
        name: mysecret
```